### PR TITLE
error handler refactoring

### PR DESCRIPTION
**Pull request by denarch22**
## 🛠️ Changes
Error handler methods are adjusted and the routes folder is deleted

## 📝 Associated issues
[Reorganización de main](https://app.zenhub.com/workspaces/all-5f7f96a92b54ca00239a9348/issues/gh/pem-humboldt/biotablero-search-backend/24)

### CODE REVIEW

File: app/utils/middleware.py
Comment (erikasv) [inline app/utils/middleware.py:1]: I'm not completely convinced of this name, it's too generic and in some cases we could end up needing a separate file for another middleware. 

This file should be named `logging_middleware` or we could create a folder `middleware` and put `logging` inside it.

Your opinions? @cazapatamar @denarch22

File: requirements.txt [hunk 1]
```diff
@@ -1,9 +1,10 @@
 fastapi>=0.110.1,<0.111.1
 pydantic-settings>=2.2.1,<2.3.0
 uvicorn[standard]>=0.29.0,<0.30.0
-black>=24.3.0,<24.4.0
+black>=24.3.0,<24.4.2
```
Comment (erikasv) [inline requirements.txt]: why this change?

File: app/utils/middleware.py
Comment (erikasv) [reply app/utils/middleware.py:1]: I just saw that fastapi [organize them](https://github.com/fastapi/fastapi/tree/master/fastapi/middleware) in the `middleware` folder, so let's go with it

File: requirements.txt [hunk 2]
```diff
@@ -1,9 +1,10 @@
...
 geojson-pydantic>=1.0.1,<1.0.2
 rio-tiler>=6.6.1,<6.7.0
 rasterio>=1.3.9,<1.3.10
 geopandas>=1.0.0,<1.0.1
 shapely>=2.0.3,<2.0.4
+starlette>=0.37.2,<0.38.0
```
Comment (erikasv) [inline requirements.txt]: Let's not include this in here, we're only using it explicitly because fastapi is based on it, so it better control those versions too

**Review by erikasv — CHANGES_REQUESTED** [review]
See the comments on each file

File: requirements.txt (hunk 1 above)
Comment (denarch22) [reply requirements.txt]: I made the change because in the review that github performs the validation of black with that new version

Comment (erikasv) [reply requirements.txt]: I see... We'll have to try to force the version we use in the action later... change the versions of the package to `>=24.4.0,<24.5.0`

**Review by erikasv — APPROVED** [review]
